libsolidity/analysis/TypeChecker.cpp

```diff
@@ -480,6 +480,10 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 				"Initial value for constant variable has to be compile-time constant."
 			);
 	}
+	else if (_variable.immutable())
+		if (_variable.type()->isValueType())
```

Comment:
The condition is inverted, though :-):
```suggestion
		if (!_variable.type()->isValueType())
```